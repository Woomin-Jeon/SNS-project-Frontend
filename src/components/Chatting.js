import React, { useState, useEffect } from 'react';
import { findUserById } from '../function';

function Chatting({
  setIsChattingOn,
  currentUserState,
  socket,
  userSocketID,
  setUserSocketID,
  loginState,
  friendID,
  setFriendID,
}) {
  const [message, setMessage] = useState('');
  const [chattingMessages, setChattingMessages] = useState([]);
  const { users } = loginState;

  useEffect(() => {
    socket.on('hello', (data) => {
      console.log('Message : ', data.message);
      setChattingMessages((chattingMessages) =>
        [...chattingMessages, { id: data.userID, msg: data.message }]);
    });
  }, []);

  const getUserInputMessage = (contents) => {
    setMessage(contents);
  };

  const sendMessage = () => {
    const userID = currentUserState.id;

    socket.emit('chat message', { message, userSocketID, userID });
    setChattingMessages([...chattingMessages, { id: currentUserState.id, msg: message }]);
    setMessage('');
  };

  const closeChatting = () => {
    setUserSocketID('');
    setFriendID('');
    setIsChattingOn(false);
  };

  return (
    <div className="chatting">
      <div className="chatting-header">
        {findUserById(users, friendID).userName}
        님 과의 채팅방입니다
      </div>
      <button className="chatting-exit" type="button" onClick={closeChatting}>X</button>
      <div>
        {chattingMessages.map((v, i) => (
          <div key={i}>
            <div className="chatting-message-body">
              <span className="chatting-partner-name">{findUserById(users, v.id).userName}</span>
              <span className="chatting-message">{v.msg}</span>
            </div>
          </div>
        ))}
      </div>
      <div className="chatting-input">
        <input type="text" value={message} onChange={(e) => getUserInputMessage(e.target.value)} />
        <button type="button" className="chatting-send-button" onClick={sendMessage}>전송</button>
      </div>
    </div>
  );
}

export default Chatting;
