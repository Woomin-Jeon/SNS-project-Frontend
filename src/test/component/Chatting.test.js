import React from 'react';
import {shallow} from "enzyme";
import Chatting from "../../components/Chatting";

describe('Snapshot test', () => {
  let currentUserState;
  let loginState;
  let setIsChattingOn;
  let userSocketID;
  let setUserSocketID;
  let friendID;
  let setFriendID;
  let socket;

  beforeEach(() => {
    loginState = {
      users: [{ id: 'Random_ID', userName: 'Woomin' }],
    };
    friendID = 'Random_ID';
  });

  it('renders well', () => {
    const component = shallow(
      <Chatting
        setIsChattingOn={setIsChattingOn}
        currentUserState={currentUserState}
        socket={socket}
        userSocketID={userSocketID}
        setUserSocketID={setUserSocketID}
        loginState={loginState}
        friendID={friendID}
        setFriendID={setFriendID}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.text()).toMatch('채팅방입니다');
    expect(component.find(".chatting-send-button").text()).toBe('전송');
  });
});
