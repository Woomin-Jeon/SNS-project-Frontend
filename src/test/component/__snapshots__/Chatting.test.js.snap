// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders well 1`] = `ShallowWrapper {}`;
